ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -11656,6 +11656,9 @@ void HqlCppTranslator::doBuildStmtOutput(BuildCtx & ctx, IHqlExpression * expr)
     if (expr->hasAttribute(groupedAtom) && (dataset->getOperator() != no_null))
         throwError1(HQLERR_NotSupportedInsideNoThor, "Grouped OUTPUT");
 
+    if (queryRealChild(expr, 1))
```

Comment:
Move the test later - so that if we are already executing within a graph it will generate a child query